Application/Do-Svyazi.User.Application/CQRS/Users/Commands/AddUser.cs

```diff
@@ -16,17 +17,23 @@ public class Handler : IRequestHandler<Command, Guid>
 
         public async Task<Guid> Handle(Command request, CancellationToken cancellationToken)
         {
-            var user = new MessengerUser
+            if (!await UserExist(request.nickName))
             {
-                Name = request.name,
-                NickName = request.nickName,
-                Description = request.description,
-            };
+                throw new Do_Svyazi_User_BusinessLogicException(
+                    $"User with nickname = {request.nickName} is already created");
+            }
 
-            await _context.Users.AddAsync(user, cancellationToken);
+            var messengerUser = new MessengerUser(request.name, request.nickName, request.description);
+            await _context.Users.AddAsync(messengerUser, cancellationToken);
             await _context.SaveChangesAsync(cancellationToken);
 
-            return user.Id;
+            return messengerUser.Id;
+        }
+
+        private async Task<bool> UserExist(string nickName)
+        {
+            MessengerUser? foundUser = await _context.Users.FindAsync(nickName);
+            return foundUser is null;
```

Comment:
`is not null` может?
```suggestion
            MessengerUser? foundUser = await _context.Users.FindAsync(nickName);
            return foundUser is not null;
```

если у тебя нашёлся юзер и ты проверил is null, то он тебе вернёт false

а метод называется UserExist (IsUserExist) -> юзер есть но он тебе вернул false